class UserInputError(Exception):
    pass

def validate_ref(ref):
    pass